storage-proofs/src/hasher/pedersen.rs

```diff
@@ -48,12 +57,48 @@ impl Hasher for PedersenHasher {
     }
 }
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct PedersenFunction(Fr);
+#[derive(Clone, PartialEq)]
+pub struct PedersenFunction {
+    // The Pedersen Hash segment index of the next unhashed segment.
+    next_segment: u32,
+    // Stores any unhashed bits between calls to `self.update()`. `unhashed_bits`'s bit-length never
+    // exceeds `N_SEGMENT_BITS` (i.e. never contains more than one segment's worth of unhashed
+    // data).
+    unhashed_bits: BitVec<LittleEndian, u8>,
+    // Stores the current Pederen Hash. A Pedersen Hash is a summation of curve points; everytime
+    // `self.update()` is called with anouther segment's worth of data, an element of the prime
+    // order JubJub subgroup (an Edward's point) is added to this value.
+    curr_hash: edwards::Point<Bls12, PrimeOrder>,
```

Comment:
We need to think about this. This is a very large element (4 Fr elements = 4 * 4 * 64bit = 128 byte). 

We will need to hold one of these in memory for each node in a sector when doing column commitments, so instead of having `NODES_COUNT * 32` we would now have `NODES_COUNT + 128` bytes in memory.

Reply:
In your last line you mean `NODE_COUNT * 128` right?

Ok, I'm not sure how to write an updateable pedersen hash without storing an entire edwards point for each node. In order to update the digest you have to do an edward's group law which uses all 3 coordinates and the twist.

We could store compressed curve points (a single Fr), but the cost of decompression is extremely high.